core/xml.js

```diff
@@ -332,6 +338,71 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
 };
 
 /**
+ * Decode an XML DOM and create blocks on the workspace.
+ * Append the new blocks at the bottom of the existing blocks
+ * @param {!Element} xml XML DOM.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ */
+Blockly.Xml.appendDomToWorkspace = function(xml, workspace) {
+  
+  // find bottom position and alignment of the current blocks  
+  //var blocks = workspace.getAllBlocks();
+  var blocks = workspace.getTopBlocks();
+  var offsetY = 0; // container for the y of the bottom block
+  var offsetX = 0; 
+  var blocksCount = blocks.length;
+  if (blocksCount > 0) {
+	// initialize the position of the corners using position 
+	// of the first block
+	var bHW = (("height" in blocks[0])? blocks[0].getHeightWidth() : {height: 0, width: 0});
```

Comment:
I am wondering what is the best solution. getBlocksBoudingBox add a Tab in case of open connection which slightly misaligned the code. As a workaround, I set Blockly.BlockSvg.TAB_WIDTH to 0 before to call the function and then back to its normal value. This help me to save a few lines of code. I hope it is ok with you.